jetstream/cli.py

```diff
@@ -323,14 +323,16 @@ def run(
     config_file,
 ):
     """Runs analysis for the provided date."""
-    AnalysisExecutor(
+    success = AnalysisExecutor(
         project_id=project_id,
         dataset_id=dataset_id,
         date=date,
         experiment_slugs=[experiment_slug] if experiment_slug else All,
         configuration_map={experiment_slug: config_file} if experiment_slug and config_file else {},
     ).execute(strategy=SerialExecutorStrategy(project_id, dataset_id))
 
+    sys.exit(0 if success else 1)
```

Comment:
This should do the trick. Every step that is launched in Argo runs the `run` command, so we could let the pods fail here. Airflow won't care and exit in a success state when run-argo is executed (it doesn't look at the status)


One thing to note here, the daily Airflow jobs also execute rerun_config_changed, so they could fail here if there is an invalid config (although that should rarely be the case so this might be ok).

Reply:
We could plausibly have something like a valid configuration that invokes a Statistic that crashes on data; I wouldn't want to retry the entire day's analyses. Is that a risk?

Reply:
Yes, that would be a risk. I added a `--return-status` flag to `rerun_config_changed`. The flag can be set when `rerun_config_changed` is invoked by jetstream-config for the status to be returned. Otherwise the actual status will be ignored.